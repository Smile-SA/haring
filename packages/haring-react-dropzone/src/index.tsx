/* eslint-disable react-refresh/only-export-components */

'use client';

// component exports
export { Dropzone } from './Components/Dropzone/Dropzone';
export type { IDropzoneProps, IFile } from './Components/Dropzone/Dropzone';
