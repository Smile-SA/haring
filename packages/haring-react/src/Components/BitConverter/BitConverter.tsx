'use client';

import type { Options } from 'pretty-bytes';
import type { ReactElement } from 'react';

import prettyBytes from 'pretty-bytes';

export interface IBitConverterProps {
  options?: Options;
  value: number;
}

export function BitConverter(props: IBitConverterProps): ReactElement {
  const { value, options } = props;

  const newValue = prettyBytes(Number(value), options);

  return <>{newValue}</>;
}
