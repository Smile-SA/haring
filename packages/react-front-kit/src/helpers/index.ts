export * from './nestedObject';
export * from './typeGuard';
