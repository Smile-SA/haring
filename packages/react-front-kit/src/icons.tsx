import type { Icon } from '@phosphor-icons/react';

import * as Icons from '@phosphor-icons/react';

const iconsToIgnore = ['IconBase', 'IconContext'];
const iconEntries = Object.entries(Icons).filter(
  ([key]) => !iconsToIgnore.includes(key)
) as [string, Icon][];

export const icons = Object.fromEntries(iconEntries);

export const iconsElements = Object.fromEntries(
  iconEntries.map(([key, Icon]) => [key, <Icon key={key} />])
);
