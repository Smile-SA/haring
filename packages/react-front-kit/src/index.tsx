/* eslint-disable react-refresh/only-export-components */
'use client';

// component exports
export type { ICardHeaderProps } from './Components/CardHeader/CardHeader';
export { CardHeader } from './Components/CardHeader/CardHeader';
export type { IMenuLanguagesProps } from './Components/MenuLanguages/MenuLanguages';
export { MenuLanguages } from './Components/MenuLanguages/MenuLanguages';
export type { IButtonListOrDropdownProps } from './Components/ButtonListOrDropdown/ButtonListOrDropdown';
export { ButtonListOrDropdown } from './Components/ButtonListOrDropdown/ButtonListOrDropdown';
export { ActionBar } from './Components/ActionBar/ActionBar';
export type { IActionBarProps } from './Components/ActionBar/ActionBar';
export { ActionRowOverflow } from './Components/ActionRowOverflow/ActionRowOverflow';
export type {
  IActionRowOverflowProps,
  IActionRowOverflowAction,
} from './Components/ActionRowOverflow/ActionRowOverflow';
export { BitConverter } from './Components/BitConverter/BitConverter';
export type { IBitConverterProps } from './Components/BitConverter/BitConverter';
export { Breadcrumbs } from './Components/Breadcrumbs/Breadcrumbs';
export type { IBreadcrumbsProps } from './Components/Breadcrumbs/Breadcrumbs';
export { CollapseButtonControlled } from './Components/CollapseButton/CollapseButtonControlled';
export type { ICollapseButtonControlledProps } from './Components/CollapseButton/CollapseButtonControlled';
export { CollapseButton } from './Components/CollapseButton/CollapseButton';
export type { ICollapseButtonProps } from './Components/CollapseButton/CollapseButton';
export { ConfirmModal } from './Components/ConfirmModal/ConfirmModal';
export type { IConfirmModalProps } from './Components/ConfirmModal/ConfirmModal';
export { DocumentBox } from './Components/DocumentBox/DocumentBox';
export type { IDocumentBoxProps } from './Components/DocumentBox/DocumentBox';
export { DocumentList } from './Components/DocumentList/DocumentList';
export type {
  IDocument,
  IDocumentListProps,
} from './Components/DocumentList/DocumentList';
export { DropdownButton } from './Components/DropdownButton/DropdownButton';
export type { IDropdownButtonProps } from './Components/DropdownButton/DropdownButton';
export { FilterList } from './Components/FilterList/FilterList';
export type { IFilterListProps } from './Components/FilterList/FilterList';
export { Header } from './Components/Header/Header';
export type { IHeaderProps } from './Components/Header/Header';
export { HeaderMobile } from './Components/HeaderMobile/HeaderMobile';
export type { IHeaderMobileProps } from './Components/HeaderMobile/HeaderMobile';
export { HeaderNav } from './Components/HeaderNav/HeaderNav';
export type {
  IHeaderNavMenu,
  IHeaderNavProps,
} from './Components/HeaderNav/HeaderNav';
export { HeaderSearch } from './Components/HeaderSearch/HeaderSearch';
export type { IHeaderSearchProps } from './Components/HeaderSearch/HeaderSearch';
export { InfoCard } from './Components/InfoCard/InfoCard';
export type {
  IContentItem,
  IInfoCardProps,
} from './Components/InfoCard/InfoCard';
export { Motif } from './Components/InfoCard/Motif';
export type { IMotifProps } from './Components/InfoCard/Motif';
export { Pagination } from './Components/Pagination/Pagination';
export type { IPaginationProps } from './Components/Pagination/Pagination';
export { Preview } from './Components/Preview/Preview';
export type { IPreviewProps } from './Components/Preview/Preview';
export { ResponsiveTabs } from './Components/ResponsiveTabs/ResponsiveTabs';
export type { IResponsiveTabs } from './Components/ResponsiveTabs/ResponsiveTabs';
export { SearchableCheckboxList } from './Components/SearchableCheckboxList/SearchableCheckboxList';
export type {
  ISearchableCheckboxListProps,
  ICheckbox,
} from './Components/SearchableCheckboxList/SearchableCheckboxList';
export { SearchBar } from './Components/SearchBar/SearchBar';
export type { ISearchBarProps } from './Components/SearchBar/SearchBar';
export { SelectableList } from './Components/SelectableList/SelectableList';
export type { ISelectableListProps } from './Components/SelectableList/SelectableList';
export { SidebarFilters } from './Components/SidebarFilters/SidebarFilters';
export type {
  ISidebarFiltersProps,
  ISidebarFilter,
} from './Components/SidebarFilters/SidebarFilters';
export {
  getRecursiveMenu,
  SidebarMenu,
} from './Components/SidebarMenu/SidebarMenu';
export type {
  IMenuItem,
  ISidebarMenuProps,
} from './Components/SidebarMenu/SidebarMenu';
export { SummaryBox } from './Components/SummaryBox/SummaryBox';
export type { ISummaryBoxProps } from './Components/SummaryBox/SummaryBox';
export { SwitchableView } from './Components/SwitchableView/SwitchableView';
export type {
  IDataView,
  ISwitchableViewProps,
} from './Components/SwitchableView/SwitchableView';
export { Thumbnail } from './Components/Thumbnail/Thumbnail';
export type { IThumbnailProps } from './Components/Thumbnail/Thumbnail';
export { ThumbnailGrid } from './Components/ThumbnailGrid/ThumbnailGrid';
export type { IThumbnailGridProps } from './Components/ThumbnailGrid/ThumbnailGrid';
// layout exports
export { FoldableColumnLayout } from './Layouts/FoldableColumnLayout/FoldableColumnLayout';
export type { IFoldableColumnLayoutProps } from './Layouts/FoldableColumnLayout/FoldableColumnLayout';
// helper exports
export {
  addPathAndDepth,
  flattenNestedObjects,
  getClosestItemFromPath,
  setChildrenToTree,
} from './helpers';
// type exports
export type { IThumbnail, IThumbnailAction, IThumbnailData } from './types';
