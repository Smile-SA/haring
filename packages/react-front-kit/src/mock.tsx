/* eslint-disable react-refresh/only-export-components */

export * from './Components/ResponsiveTabs/ResponsiveTabs.mock';
export * from './Components/SidebarMenu/SidebarMenu.mock';
export * from './Components/SwitchableView/SwitchableView.mock';
export * from './Components/Thumbnail/Thumbnail.mock';
export * from './Components/ThumbnailGrid/ThumbnailGrid.mock';
export * from './Layouts/FoldableColumnLayout/FoldableColumnLayout.mock';
