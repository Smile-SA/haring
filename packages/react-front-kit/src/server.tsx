export * from '@smile/react-front-kit-shared/server';
// other exports
export * from './helpers';
