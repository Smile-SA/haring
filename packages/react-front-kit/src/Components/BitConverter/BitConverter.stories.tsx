import type { Meta, StoryObj } from '@storybook/react';

import { BitConverter as Cmp } from './BitConverter';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/BitConverter',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const BitConverter: IStory = {
  args: {
    options: { locale: 'fr' },
    value: 121212,
  },
};
