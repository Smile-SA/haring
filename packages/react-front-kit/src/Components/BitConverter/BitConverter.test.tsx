import { renderWithProviders } from '@smile/react-front-kit-shared/src/test-utils';

import { BitConverter } from './BitConverter';

describe('BitConverter', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <BitConverter options={{ locale: 'fr' }} value={12121212} />,
    );
    expect(container).toMatchSnapshot();
  });
});
