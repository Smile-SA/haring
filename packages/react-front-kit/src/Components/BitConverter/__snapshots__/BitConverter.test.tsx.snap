// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BitConverter matches snapshot 1`] = `
<div>
  12,1 MB
</div>
`;
